app/models/ems_refresh/refreshers/ems_refresher_mixin.rb

```diff
@@ -43,6 +47,7 @@ def refresh
         end
 
         _log.info "Refreshing all targets...Complete"
+        raise PartialRefreshError if partial_refresh_error
```

Comment:
> who catches this? the miq_queue deliver rescue?

yes, that's the idea, so the queue message is delivered in error and can set the state to `error`

>  Does it make sense to give this exception a useful error message as to what provider refresh failed and what error it was?

Absolutely, this was just to flesh things out and check if this is the right approach :smile: